lib/auth/webauthncli/fido2.go

```diff
@@ -128,28 +139,32 @@ func fido2Login(
 	var usedAppID bool
 
 	pathToRPID := &sync.Map{} // map[string]string
-	filter := func(dev FIDODevice, info *deviceInfo) (bool, error) {
+	filter := func(dev FIDODevice, info *deviceInfo) error {
 		switch {
+		case info.u2f && (uv || passwordless):
+			return errPasswordlessU2F
+		case passwordless && (!info.uvCapable() || !info.rk):
+			return errNoPasswordless
```

Comment:
If a device doesn't have the necessary capabilities it cannot have been registered, hence the slight cognitive jump from `if` condition to the end message.